File: iCinema.Application/DTOs/Reports/ReportResponseDto.cs
namespace iCinema.Application.DTOs.Reports;

public class ReportResponseDto
{
    public ReportType ReportType { get; set; }
    public ReportPeriodDto Period { get; set; } = new();
    public List<Dictionary<string, object>> Data { get; set; } = new();
}

public class ReportPeriodDto
{
    public DateTime From { get; set; }
    public DateTime To { get; set; }
}

File: iCinema.UI/icinema_desktop/lib/features/reports/data/report_response_dto.dart
class ReportResponseDto {
  final int reportType;
  final ReportPeriodDto period;
  final List<Map<String, dynamic>> data;

  ReportResponseDto({
    required this.reportType,
    required this.period,
    required this.data,
  });

  factory ReportResponseDto.fromJson(Map<String, dynamic> json) {
    return ReportResponseDto(
      reportType: json['reportType'] as int,
      period: ReportPeriodDto.fromJson(json['period'] as Map<String, dynamic>),
      data: (json['data'] as List<dynamic>)
          .map((item) => item as Map<String, dynamic>)
          .toList(),
    );
  }
}

class ReportPeriodDto {
  final DateTime from;
  final DateTime to;

  ReportPeriodDto({
    required this.from,
    required this.to,
  });

  factory ReportPeriodDto.fromJson(Map<String, dynamic> json) {
    return ReportPeriodDto(
      from: DateTime.parse(json['from'] as String),
      to: DateTime.parse(json['to'] as String),
    );
  }
}
